File: dmn-core/src/test/resources/tck/cl2/expected/dmn/0008-lx-arithmetic/type/TLoanImpl.java
package type;

import java.util.*;

@javax.annotation.Generated(value = {"itemDefinition.ftl", "tLoan"})
@com.fasterxml.jackson.annotation.JsonPropertyOrder(alphabetic = true)
public class TLoanImpl implements TLoan {
        private java.math.BigDecimal principal;
        private java.math.BigDecimal rate;
        private java.math.BigDecimal termMonths;

    public TLoanImpl() {
    }

    public TLoanImpl(java.math.BigDecimal principal, java.math.BigDecimal rate, java.math.BigDecimal termMonths) {
        this.setPrincipal(principal);
        this.setRate(rate);
        this.setTermMonths(termMonths);
    }

    @com.fasterxml.jackson.annotation.JsonGetter("principal")
    public java.math.BigDecimal getPrincipal() {
        return this.principal;
    }

    @com.fasterxml.jackson.annotation.JsonSetter("principal")
    public void setPrincipal(java.math.BigDecimal principal) {
        this.principal = principal;
    }

    @com.fasterxml.jackson.annotation.JsonGetter("rate")
    public java.math.BigDecimal getRate() {
        return this.rate;
    }

    @com.fasterxml.jackson.annotation.JsonSetter("rate")
    public void setRate(java.math.BigDecimal rate) {
        this.rate = rate;
    }

    @com.fasterxml.jackson.annotation.JsonGetter("termMonths")
    public java.math.BigDecimal getTermMonths() {
        return this.termMonths;
    }

    @com.fasterxml.jackson.annotation.JsonSetter("termMonths")
    public void setTermMonths(java.math.BigDecimal termMonths) {
        this.termMonths = termMonths;
    }

    @Override
    public boolean equals(Object o) {
        return equalTo(o);
    }

    @Override
    public int hashCode() {
        return hash();
    }

    @Override
    public String toString() {
        return asString();
    }
}

File: dmn-test-cases/tck/1.1/cl2/0008-LX-arithmetic/translator/expected/kotlin/dmn/type/TLoanImpl.kt
package type

import java.util.*

@javax.annotation.Generated(value = ["itemDefinition.ftl", "tLoan"])
@com.fasterxml.jackson.annotation.JsonPropertyOrder(alphabetic = true)
class TLoanImpl : TLoan {
    @get:com.fasterxml.jackson.annotation.JsonGetter("principal")
    @set:com.fasterxml.jackson.annotation.JsonGetter("principal")
    override var principal: java.math.BigDecimal? = null

    @get:com.fasterxml.jackson.annotation.JsonGetter("rate")
    @set:com.fasterxml.jackson.annotation.JsonGetter("rate")
    override var rate: java.math.BigDecimal? = null

    @get:com.fasterxml.jackson.annotation.JsonGetter("termMonths")
    @set:com.fasterxml.jackson.annotation.JsonGetter("termMonths")
    override var termMonths: java.math.BigDecimal? = null

    constructor() {
    }

    constructor (principal: java.math.BigDecimal?, rate: java.math.BigDecimal?, termMonths: java.math.BigDecimal?) {
        this.principal = principal
        this.rate = rate
        this.termMonths = termMonths
    }

    override fun equals(other: Any?): Boolean {
        return equalTo(other)
    }

    override fun hashCode(): Int {
        return hash()
    }

    @Override
    override fun toString(): String {
        return asString()
    }
}
